pkg/api/controller.go

```diff
@@ -263,6 +263,144 @@ func (c *Controller) CreatePresignMultipartUpload(w http.ResponseWriter, r *http
 	writeResponse(w, r, http.StatusCreated, resp)
 }
 
+func (c *Controller) UploadPartFrom(w http.ResponseWriter, r *http.Request, body apigen.UploadPartFromJSONRequestBody, dstRepository string, branch string, uploadID string, partNumber int, params apigen.UploadPartFromParams) {
+	isCopy := strings.ToLower(body.Type) == "copy"
```

Comment:
if we are very explicit in the interface - don't use to lower and fail if not empty (default) or match the exact type.